fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/settings/SettingsActivity.java

```diff
@@ -0,0 +1,71 @@
+package com.github.dedis.popstellar.ui.settings;
+
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.FragmentActivity;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.github.dedis.popstellar.R;
+import com.github.dedis.popstellar.ViewModelFactory;
+import com.github.dedis.popstellar.utility.ActivityUtils;
+
+public class SettingsActivity extends AppCompatActivity {
+
+  private SettingsViewModel mViewModel;
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.settings_activity);
+
+    setupSettingsFragment();
+
+    mViewModel = obtainViewModel(this);
+
+    // Subscribe to "open settings" event
+    mViewModel
+        .getOpenSettingsEvent()
+        .observe(
+            this,
+            booleanEvent -> {
+              Boolean event = booleanEvent.getContentIfNotHandled();
+              if (event != null) {
+                setupSettingsFragment();
+              }
+            });
+  }
+
+  @Override
+  public boolean onCreateOptionsMenu(Menu menu) {
+    getMenuInflater().inflate(R.menu.options_menu, menu);
+    return true;
+  }
+
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    if (item.getItemId() == R.id.settings) {
+      mViewModel.openSettings();
+      return true;
+    } else {
+      return super.onOptionsItemSelected(item);
+    }
+  }
```

Comment:
If we are already in the settings activity, having the settings menu is maybe not very usefull.
```suggestion
  }
```